### PR TITLE
[addons/karpenter] Covers the use of KubernetesVersion.of('1.28') syntax to get a KubernetesVersion

### PR DESCRIPTION
**Pull request by ROunofF**
*Issue #, if available:* #976

*Description of changes:* This is a test that reproduce the problem I faced in #976, I'll look at possible solution next. It's currently failing (hence the bug). Let me know if you think that is wrong.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/karpenter.test.ts [hunk 1]
```diff
@@ -428,4 +428,28 @@ describe('Unit tests for Karpenter addon', () => {
...
         expect(manifest.spec.blockDeviceMappings[0]).toMatchObject(blockDeviceMapping);
       });
+
+      test("Stack creation succeeded using KubernetesVersion.Of", () => {
+        const app = new cdk.App();
+
+        const blueprint = blueprints.EksBlueprint.builder();
+
+        blueprint.account("123567891").region('us-west-1')
+            .version(KubernetesVersion.of('1.28'))
```
Comment (elamaran11) [inline test/karpenter.test.ts:438]: Why are we doing `.of`? Can we not use `eks.KubernetesVersion.V1_28` ?

**Review by elamaran11 — COMMENTED** [review]
Thanks for the PR. Minor comment!

File: test/karpenter.test.ts (hunk 1 above)
Comment (ROunofF) [reply test/karpenter.test.ts:438]: It should be able possible to use both style... This PR allows both ways to set the version.

**Review by elamaran11 — COMMENTED** [review]
@ROunofF My question though is why is this even required? Can we not use `KubernetesVersion.V1_2X`
@shapirov103 Thoughts?

**Pull request merged**